spring-integration-smb/src/main/java/org/springframework/integration/smb/gateway/SmbOutboundGateway.java

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.smb.gateway;
+
+import java.net.MalformedURLException;
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.expression.Expression;
+import org.springframework.expression.spel.support.StandardEvaluationContext;
+import org.springframework.integration.expression.ExpressionUtils;
+import org.springframework.integration.file.remote.AbstractFileInfo;
+import org.springframework.integration.file.remote.MessageSessionCallback;
+import org.springframework.integration.file.remote.RemoteFileTemplate;
+import org.springframework.integration.file.remote.gateway.AbstractRemoteFileOutboundGateway;
+import org.springframework.integration.file.remote.session.SessionFactory;
+import org.springframework.integration.smb.session.SmbFileInfo;
+import org.springframework.integration.smb.session.SmbRemoteFileTemplate;
+
+import jcifs.smb.SmbException;
+import jcifs.smb.SmbFile;
+
+/**
+ * Outbound Gateway for performing remote file operations via SMB.
+ *
+ * @author Gregory Bragg
+ *
+ * @since 6.0
+ */
+public class SmbOutboundGateway extends AbstractRemoteFileOutboundGateway<SmbFile> {
+
+	private static final Log logger = LogFactory.getLog(SmbOutboundGateway.class);
+
+	private Expression workingDirExpression;
+
+	private StandardEvaluationContext evaluationContext;
```

Comment:
Yeah I know it's weird... my IDE says the same thing, but wehn I remove them I get a compile error. They are both used in
`setWorkingDirExpression(Expression workingDirExpression)` and `doInit()`

Reply:
Correct. That's settings values to them. But they are out of use afterwards in the logic.
See for example `FtpOutboundGateway.doInWorkingDirectory()`

Reply:
Yeah ok, see it... I think those methods are redundant then, because I stripped out the logic for FTP style logic such as ls, nlst, get, etc. I don't think it's really applicable to SMB protocol.